1.0 release changes

adamyanalunas opened the pull request:
Going against any reasonable amount of restraint or focus I present the 1.0 candidate PR for loudspeaker.

Logic changes, architectural changes, properties and methods removed, properties and methods added, features added, accessibility enabled, new image assets; this PR has just about every type of change that should be in individual PRs—but all in one!

Good luck. 🎉

adamyanalunas committed:
Update example project with icons, launch storyboard, drop iOS 8

adamyanalunas committed:
iPhone X support, 1.0 release changes

Restructuring, better ownership of logic, configuration for view.
Added methods, removed properties, droppied iOS 8 support.
So much more. Read the README for all details.

SergeyKuryanov commented on Example/Tests/LSPProgressViewSpec.m:
```diff
@@ -0,0 +1,158 @@
+
+SpecEnd
+
+//@interface LSPProgressViewSpecHelper : NSObject
```
Leftover?

SergeyKuryanov commented on Example/loudspeaker/LSPViewController.m:
```diff
@@ -36,6 +38,13 @@ - (IBAction)launchDemo:(id)sender
 }
 
 
+- (void)closeDemo
```
Bit confusing name, as method don't close anything.

SergeyKuryanov commented on Example/loudspeaker/LSPViewController.m:
```diff
@@ -46,100 +55,55 @@ - (void)configureDemo:(id)sender
-        self.audioVC.view.playbackTimeLabel.textColor = UIColor.brownColor;
-        self.audioVC.view.titleLabel.textColor = UIColor.yellowColor;
+        self.configuration = [LSPConfigurationBuilder configurationWithBuilder:^(LSPConfigurationBuilder *builder) {
+            builder.height = 120;
```
Why do we need set height in configuration? Why not let users specify player frame?

adamyanalunas replied:
Good point. Will change.

SergeyKuryanov commented on Pod/Classes/LSPAudioPlayerModel.h:
```diff
@@ -0,0 +1,24 @@
+//
+//  LSPAudioPlayerModel.h
+//  Expecta
```
What does Expecta mean?

adamyanalunas replied:
Xcode issue. Still belongs to the `loudspeaker` bundle. Will resolve.

SergeyKuryanov commented on Pod/Classes/LSPAudioViewController.m:
```diff
@@ -108,24 +118,75 @@ - (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(N
+}
+
+
+- (void)hide:(nullable void (^)(void))completion
```
`nullable` annotation in `m` file.

SergeyKuryanov commented on Pod/Classes/LSPAudioViewController.m:
```diff
@@ -108,24 +118,75 @@ - (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(N
+- (void)hide:(nullable void (^)(void))completion
+{
+    __weak typeof(self) weakSelf = self;
+    [weakSelf.view layoutIfNeeded];
```
Why use `weakSelf` here?

adamyanalunas replied:
Figured if I needed to declare a `weakSelf` I may as well use it as long as I've got it. Will use `self.view` and declare the weak version closer to the block.

SergeyKuryanov commented on Pod/Classes/LSPAudioViewController.m:
```diff
@@ -108,24 +118,75 @@ - (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(N
+    UIViewAnimationOptions options = (UIViewAnimationOptionBeginFromCurrentState | UIViewAnimationOptionCurveEaseInOut);
+    
+    [UIView animateWithDuration:duration delay:delay options:options animations:^{
+        __strong typeof(weakSelf) strongSelf = weakSelf;
```
Why use `strongSelf` here?

SergeyKuryanov commented on Pod/Classes/LSPAudioViewController.m:
```diff
@@ -108,24 +118,75 @@ - (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(N
+    NSTimeInterval delay = 0;
+    UIViewAnimationOptions options = (UIViewAnimationOptionBeginFromCurrentState | UIViewAnimationOptionCurveEaseInOut);
+    
+    [UIView animateWithDuration:duration delay:delay options:options animations:^{
```
I think no reference cycles can appear in this block, so `weakify` is redundant here.

SergeyKuryanov reviewed (changes requested):
Lots of useful changes. Nice work and congrats with version 1.0

For iPhone X Apple recommend extend bottom view's background to the end of the screen. Can be done in another PR.

Also when using example HarleQuin player, which have bigger height. Track title, playback bar and time labels are not centered. Is this expected?

adamyanalunas committed:
Update player size to use parent frame, clean up dead code

adamyanalunas commented:
@SergeyKuryanov I believe I addressed all of your points. The layout does not degrade gracefully when using a non-standard height but I figured that could come in another PR. Maybe now that this only supports iOS 9 things could be changed to use `UIStackView` or the auto layout rules updated to be more flexible.

SergeyKuryanov reviewed (approved):
Looks good 👍
